src/util/FS.cpp

```diff
@@ -3,18 +3,17 @@
 #include <sys/stat.h>
 #include <fstream>
 #include <sstream>
+#include <filesystem>
 
 namespace util {
 
 const bool FS::FileExists( const string& filename ) {
-	struct stat buffer;   
-	return ( stat ( filename.c_str(), &buffer ) == 0 ); 
+	return std::filesystem::exists(filename);
 }
 
 void FS::CreateDirectoryIfNotExists( const string& path ) {
-	struct stat st = {0};
-	if ( stat( path.c_str(), &st ) == -1) {
-		mkdir( path.c_str(), 0700);
+	if (!std::filesystem::is_directory(path) || !std::filesystem::exists(path)) {
+		std::filesystem::create_directory(path);
```

Comment:
I added the cross platform way of doing it here. If you want I could implement a windows specific way and have an #ifdef here. Otherwise we can have it like this